src/main/scala/com/gu/emailservices/EmailFields.scala

```diff
@@ -1,35 +1,39 @@
 package com.gu.emailservices
 
-import io.circe.{Json, Printer}
 import org.joda.time.DateTime
 import org.joda.time.format.DateTimeFormat
+import io.circe.generic.auto._
+import io.circe.syntax._
+
+case class EmailPayloadContactAttributes(SubscriberAttributes: Map[String, String])
```

Comment:
Do you think these classes are adding anything? Personally I prefer being able to see the full structure of the final json rather than having to infer it from a class hierarchy. Were you looking for improved compile time verification or something?

Reply:
doing your own serialisation can look nice, but it can be quite dangerous because now you are responsible for all your own escaping.  a hopefully well tested library to do the serialisation to string is safer.  I think 3 of the top 10 including number 1 OWASP top 10 are caused by this kind of thing.

Reply:
The advantage is compile time type safety. You're less likely to make mistakes by relying on the circe serializer typeclasses compared to building the json string dynamically using custom logic (I don't want to write code for converting optional fields to json!).

I don't see the difficulties reading a simple case class hierarchy. The json conversion is trivial.